addons/pos_paytm/static/src/js/payment_paytm.js

```diff
@@ -0,0 +1,159 @@
+/** @odoo-module */
+
+import { PaymentInterface } from "@point_of_sale/app/payment/payment_interface";
+import { ErrorPopup } from "@point_of_sale/app/errors/popups/error_popup";
+
+const REQUEST_TIMEOUT = 5000
+
+export class PaymentPaytm extends PaymentInterface {
+
+    /**
+    * @Override
+    * @param { string } cid
+    * @returns Promise
+    */
+    async send_payment_request(cid) {
+        await super.send_payment_request(...arguments);
+        const paymentLine = this.pos.get_order()?.selected_paymentline;
+        const order = this.pos?.selectedOrder;
+        let retry = localStorage.getItem(order.uid) || 0;
+        if (retry === 0) {
+            localStorage.setItem(order.uid, retry);
```

Comment:
https://github.com/odoo/odoo/pull/126175#discussion_r1246518015

[davidmonnom](https://github.com/davidmonnom) [on Jun 29](https://github.com/odoo/odoo/pull/126175#discussion_r1246518015)
You can make a function that manages the retry counter, to avoid having pieces of localstorage read/write code all over the place.